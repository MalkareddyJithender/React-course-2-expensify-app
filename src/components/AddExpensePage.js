import React from 'react';
import {connect} from 'react-redux';
import {startAddExpense,addExpense} from '../actions/expenses';
import ExpenseForm from './ExpenseForm';

//to avoid inline functions we are creating class based component

export class AddExpensePage extends React.Component
{
    onSubmit = (expense) => 
    {
    this.props.startAddExpense(expense);
    this.props.history.push('/');
    };

    render()
    {
        return (
            <div>
                <h1>Add Expense</h1>
                <ExpenseForm 
                onSubmit={this.onSubmit}
                />
            </div>
        );
    };
}


const mapDispatchToProps = (dispatch) =>({ startAddExpense:(expense) => dispatch(startAddExpense(expense))});

export default connect(undefined,mapDispatchToProps)(AddExpensePage);