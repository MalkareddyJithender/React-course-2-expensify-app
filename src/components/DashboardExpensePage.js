import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';

const DashboardExpensePage = () =>  (
    <div>
       <ExpenseListFilters />
       <ExpenseList />
    </div>
);


export default DashboardExpensePage;