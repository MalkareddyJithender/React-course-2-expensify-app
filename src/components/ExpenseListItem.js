import React from 'react';
import {Link} from 'react-router-dom';
import moment from 'moment';
import currencyFormatter from 'currency-formatter';



const ExpenseListItem = ({id,description,amount,createdAt}) => (
    <div>
        <Link to={`/edit/${id}`}>
        <h3>{description}</h3>
        </Link>
        <p>
            {currencyFormatter.format(amount/100,{code:'INR'})}            
              -     
            {moment(createdAt).format('MMMM Do, YYYY')}
             </p>
    </div>
);

export default ExpenseListItem;