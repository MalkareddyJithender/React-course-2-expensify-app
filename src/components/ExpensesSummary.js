import React from 'react';
import {connect} from 'react-redux';
import { Link } from 'react-router-dom';
import currencyFormatter from 'currency-formatter';
import getVisibleExpenses from '../selecters/expenses';
import getExpensesTotal from '../selecters/expenses-total';



export class ExpensesSummary extends React.Component
{
    render()
    {
        const expenseWord = this.props.expensesCount === 1 ? 'expense' : 'expenses';
        return (
            <div className="page-header">
                <div className="content-container">
                {
                <h1 className="page-header__title">
                 viewing <span>{this.props.expensesCount}</span> {expenseWord} totalling  <span>{ currencyFormatter.format(this.props.expensesTotal/100,{code:'INR'})}</span> 
                </h1> 
                }
                <div className="page-header__actions">
                <Link className="button" to="/create">Add Expense</Link>
                </div>
                </div>
            </div>
        );
    }
};

const mapStateToProps = (state) =>
{
    const visibleExpenses = getVisibleExpenses(state.expenses,state.filters);
    return {
        expensesCount:visibleExpenses.length,
        expensesTotal:getExpensesTotal(visibleExpenses)
    }
};

export default connect(mapStateToProps)(ExpensesSummary);