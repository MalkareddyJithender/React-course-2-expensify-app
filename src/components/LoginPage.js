import React from 'react';
import { startLogin } from '../actions/auth';
import { connect } from 'react-redux';

export class LoginPage extends React.Component
{
    render()
    {
        return (
            <div>
                <button onClick={this.props.startLogin} >Login</button>
            </div>
        );
    };
};

const mapDispatchToProps = (dispatch) =>({
    startLogin:() => dispatch(startLogin())
});

export default connect(undefined,mapDispatchToProps)(LoginPage);