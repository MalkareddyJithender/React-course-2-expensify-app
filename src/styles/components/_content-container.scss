.content-container
{
    margin:0 auto;
    max-width:80rem;
    padding:0 $m-size;
}

.content-container--profile
{
    @extend .content-container;
    max-width:90rem;
    margin-top:$s-size;
    padding:$s-size 0;
    border:1px solid lighten($grey,50);

    @media(max-width:$desktop-breakpoint - 1)
    {
        margin:$s-size $s-size 0 $s-size;
    }
}

.content-container--About
{
    @extend .content-container;
    max-width:90rem;
    padding:0;
    margin-top:$m-size;
    border:1px solid lighten($grey,50);

    @media(max-width:$desktop-breakpoint - 1)
    {
        margin:$s-size $s-size $s-size $s-size;
    }
}