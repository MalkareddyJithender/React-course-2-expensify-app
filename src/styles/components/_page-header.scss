.page-header
{
    background:$off-white;
    padding:$m-size 0;
    margin-bottom:$l-size;
}

.page-header__title
{
    margin:0;
    font-weight:300;
    span{
        font-weight:700;
    }
}

.page-header__subtitle
{
    font-weight:300;
    font-family:sans-serif;
    span{
        font-weight:700;
    }
    margin:$s-size;
}

.page-header__actions
{
    margin-top:$m-size;
}

.page-header__user
{
    display:flex;
    justify-content: space-between;
    align-items:center;
}

.page-header__user__name
{
    margin:0;
    font-weight:300;
    font-size:$font-size-large;
    font-family:verdana;
}

.page-header__user__profile
{
    text-decoration:none;
    color:green;
    margin-right:$l-size;
}

#name
{
    color:green;
    font-weight:400;
}