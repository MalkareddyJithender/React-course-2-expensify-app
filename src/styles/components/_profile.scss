.img
{
    height:15rem;
    width:15rem;
    border:none;
    border-radius:100%;
}

.i-icon
{
    height:1.3rem;
    width:1.3rem;
    border-radius: 100%;
    margin-left:8px;
    margin-bottom:2px;
}

#profile
{
    display:flex;
    justify-content:space-evenly;
    flex-direction:row;

    @media(max-width:$desktop-breakpoint)
    {
        flex-direction:column;
        justify-content:center;
        align-items:center;
    }
}

.title
{
    margin:4.5rem 0 0 0;
    font-weight:300;
    font-family:sans-serif;
    color:rgb(165, 165, 165);

    @media(max-width:$desktop-breakpoint - 1)
    {
        margin-top:1rem;
    }
}

.About__title
{
    display: flex;
    align-items: center;
    background:$blue;
    color:white;
    font-weight:300;
    border-radius:1px;
}

.About__title__2
{
    margin:0;
}

.About__content
{
    margin:$m-size;
    font-weight:300;
}

.About__content__header
{
    margin:0;
    font-weight:100;
    font-size:1.5rem;
    color:ligthen($grey,100%);
}

.About__content__body
{
    margin:0;
    font-weight:700;
    font-size:$font-size-small;
    border-bottom:1px solid lighten($grey,50%);
}
