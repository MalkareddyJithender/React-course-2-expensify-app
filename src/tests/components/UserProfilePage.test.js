import React from 'react';
import { shallow } from 'enzyme';
import { UserProfilePage } from '../../components/UserProfilePage';

test('should render UserProfilePage component correctly',() =>
{
    const auth =
    {
        uid:'abc123',
        userName:'Malkareddy Jithender',
        userEmail:'jithendermalkareddy@gmail.com',
        userPhoto:'sbxjkbjkbsadckcnkncknjnjjjkl',
        userPhone:'+916302696865'
    };
    const wrapper = shallow(<UserProfilePage auth={auth} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render UserProfilePage component with no values',() =>
{
    const auth =
    {
        uid:'abc123',
        userName:'',
        userEmail:'',
        userPhoto:'',
        userPhone:''
    };
    const wrapper = shallow(<UserProfilePage auth={auth} />);
    expect(wrapper).toMatchSnapshot();
});

