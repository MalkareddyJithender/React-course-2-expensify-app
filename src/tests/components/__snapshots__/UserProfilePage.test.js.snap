// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render UserProfilePage component correctly 1`] = `
<div>
  <div
    className="content-container--profile"
  >
    <div
      id="profile"
    >
      <img
        className="img"
        src="sbxjkbjkbsadckcnkncknjnjjjkl"
      />
      <div>
        <h1
          className="title"
        >
          User Profile
        </h1>
      </div>
    </div>
  </div>
  <div
    className="content-container--About"
  >
    <div
      className="About__title"
    >
      <img
        className="i-icon"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAOEAAADhCAMAAAAJbSJIAAAAeFBMVEX///8AAADExMTw8PD7+/v8/Pzb29vo6Oj09PTBwcGfn5/h4eFpaWlvb2/4+Pjt7e0dHR0zMzMPDw+lpaXOzs6NjY0kJCRcXFxDQ0OVlZVTU1MTExOqqqq7u7s3NzdKSkqDg4N3d3crKytgYGCHh4c/Pz+zs7NPT096BE0EAAAGMElEQVR4nO2d23aCOhCGCcjBooCIVsQDaGvf/w23Wg8ImYCFvbJ/9nyXtRfzr5DMZDKTGAbDMAzDMAzDMAzDMAzDMAzDMJ0Ik1OabsyRbjv+HQJzGokb8cS1ddvTN9ZmJl5Ym4Fum/okMH9EjYWr26z+sNJlXaAQE1+3YX0RTmX6znyGuk3rB29NCDxL1G1bP6xIgULkuo3rg0IhUCzHus3rzilWKRQFvF/090qBYgY/iAu1QCE+dFvYEadhCM+fqaXbxk7YH00CxY+j28hO+MdGhZmp28hOJI0CRQyt0N4MXaFFBaSDUeiqArZhKIyGrtBvFihmiW4ru9BGIbY/bKMwh07XtFE4121kJ1oo3EEvNIZ7aFQ48XQb2Ynws0lgDL55CtImhSvsITQMs0nhRreFXfElqe4yA8gmzpUCDwPI6zvKQdzqNq8PNnTwnc3hM4kXgoISGA9DoGGMqFQNuCcsIz2YOQxiDt75qgVv0WQg52p3kuLliPswhXf0NQIzPf4eA8ezSTqG3hJS2H4y3p4xkwF4eYZhGIZhGIZhGIb53+O5vnPGH163xQUvMTf55y7Osmg1+TAHlmwzbGeTv5QPZcV2SPmo0bao15kuc+zK0jLzH2lDiZgOZBRP9OnTEI4ugq18+G7A505HSUP5ZaHbwo648oauEpluE7uRTBr0CREjd6/Z8+9GgUIA95OEubob6Abuauq0GUCBW41hj9vpEzFo+aylrhMqEek29W/4RVuB4lu3rX8iaSy5fALp8ZOWa8wVxALhbYsWiye6rf0DrdeYK0vd5r5N8PWWQLHQbfC7uMV7AgVau4zbHGlXAMtGOc39ohXWWGXsfotOvApYBaYh0TzyvaB3wVDTMCRETL2CFLhH2v4SfRXxIgjoGO4LZxraY/knGp0n2pZufTrptrs19ngnVZBdtrd0XxBQ82giF7i/hNWKbu4JTLKUcBPr6xCZM+mPF1LdhrclJM4kfkfoRArMUGq9Lbn9+e86aeWkwjWIr7Dkt1zdNw0OvZKCbO9DabCdPaYYnXPLQDKl0g3vc4YF9A1DM5A8oiuRsHx6csW9GHuNVr9FUPsOv0vZJYcUKHJdFr+P/+otduVIRdHojBPQGIY3LZ3BrMqW27S7xzo4DOYPn7B+WT8IX3lhosvYP3K/nfTzNdRU3KEElwp2rgF29T5SxV1teDcJh2ldYEALROzDD7ZFNdBUHCKiRN0NKCpNYLaGauhDqD1YKpjApNOIH3gLjQzFvZ5IAY0C+hRjBbKvaEBxcylQolSF4soPnESpEnpfsRvGNFTczQp+r+4dla/QbVs/0PuKgdxqokqUDiNkc+gcFEiitAn6iuQY5rxCyYi+x/swDF9h0Se/MIlSNS4pUOS6besHxX2JwLXrZRTbe6gaGho66l7pNq0fPHoIBxKyKapMhxF1K1LBkSRDgzgz6XYZycM53gkvqaF41aJ+XhEuxB5OouLYsLavCIu4emQFAL29r6WCg9+MHJhEj945Laq9zfdiFazcjaL1t6KjVJWaIUVzinLE14/RL5f8RUASC1LhtLzQ2Mmr28SRqLhM/yVkM6tnUzAvP5nyctMLJ/W/obwytyEjmllpkOay5Qgk0UgvNMfnNJTH5jOI71TRG/Ro5BrJO8BAnGJC+/v7QkP0YKwwBBrjjBJ4n2VEi9snSuRGV3bfemSIz/gIUhWtenjleP0HR143fUQZQcOl/f31wMIlSvtxshsOGXZfnwXayn9DKlWky4IjkzzOmOq2+h3oYrZD6BElNlANz4rq/J1VSP+eYR23eXREs5T/tNtglYBZdGm3nMMWLF1KdcuSAwsSqT1RHBxKBeJVLbR44rjED0qkVqLFE8dPjjiBzJPG5wBL5JCVwi0ecb4RLzBrMNtf/jEHvdqztUKIjIyMlgojmN1gjXYK93hu8IGiQP9BXAALpHa4ZbIFpJe40/yKc3TC9BJ3LPrQ4vaJOmB7iSqKsssrYFdDyVDconB5W1W3eT0wUtRDrWHd/AsuFZquUmQnUUae1I4WcLt5Gr8e2BzSBDTQluNV5uJu7g9K3wXv45Fzm+U+uAekGDmbND0BJmIYhmEYhmEYhmEYhmEYhmEY5j/FPx3KSD8WrnwpAAAAAElFTkSuQmCC"
      />
        
      <h5
        className="About__title__2"
      >
        About
      </h5>
    </div>
    <div
      className="About__content"
    >
      <p
        className="About__content__header"
      >
        Name
      </p>
      <p
        className="About__content__body"
      >
        Malkareddy Jithender
      </p>
    </div>
    <div
      className="About__content"
    >
      <p
        className="About__content__header"
      >
        Email
      </p>
      <p
        className="About__content__body"
      >
        jithendermalkareddy@gmail.com
      </p>
    </div>
    <div
      className="About__content"
    >
      <p
        className="About__content__header"
      >
        Mobile Phone
      </p>
      <p
        className="About__content__body"
      >
        +916302696865
      </p>
    </div>
  </div>
</div>
`;

exports[`should render UserProfilePage component with no values 1`] = `
<div>
  <div
    className="content-container--profile"
  >
    <div
      id="profile"
    >
      <img
        className="img"
        src=""
      />
      <div>
        <h1
          className="title"
        >
          User Profile
        </h1>
      </div>
    </div>
  </div>
  <div
    className="content-container--About"
  >
    <div
      className="About__title"
    >
      <img
        className="i-icon"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAOEAAADhCAMAAAAJbSJIAAAAeFBMVEX///8AAADExMTw8PD7+/v8/Pzb29vo6Oj09PTBwcGfn5/h4eFpaWlvb2/4+Pjt7e0dHR0zMzMPDw+lpaXOzs6NjY0kJCRcXFxDQ0OVlZVTU1MTExOqqqq7u7s3NzdKSkqDg4N3d3crKytgYGCHh4c/Pz+zs7NPT096BE0EAAAGMElEQVR4nO2d23aCOhCGCcjBooCIVsQDaGvf/w23Wg8ImYCFvbJ/9nyXtRfzr5DMZDKTGAbDMAzDMAzDMAzDMAzDMAzDMJ0Ik1OabsyRbjv+HQJzGokb8cS1ddvTN9ZmJl5Ym4Fum/okMH9EjYWr26z+sNJlXaAQE1+3YX0RTmX6znyGuk3rB29NCDxL1G1bP6xIgULkuo3rg0IhUCzHus3rzilWKRQFvF/090qBYgY/iAu1QCE+dFvYEadhCM+fqaXbxk7YH00CxY+j28hO+MdGhZmp28hOJI0CRQyt0N4MXaFFBaSDUeiqArZhKIyGrtBvFihmiW4ru9BGIbY/bKMwh07XtFE4121kJ1oo3EEvNIZ7aFQ48XQb2Ynws0lgDL55CtImhSvsITQMs0nhRreFXfElqe4yA8gmzpUCDwPI6zvKQdzqNq8PNnTwnc3hM4kXgoISGA9DoGGMqFQNuCcsIz2YOQxiDt75qgVv0WQg52p3kuLliPswhXf0NQIzPf4eA8ezSTqG3hJS2H4y3p4xkwF4eYZhGIZhGIZhGIb53+O5vnPGH163xQUvMTf55y7Osmg1+TAHlmwzbGeTv5QPZcV2SPmo0bao15kuc+zK0jLzH2lDiZgOZBRP9OnTEI4ugq18+G7A505HSUP5ZaHbwo648oauEpluE7uRTBr0CREjd6/Z8+9GgUIA95OEubob6Abuauq0GUCBW41hj9vpEzFo+aylrhMqEek29W/4RVuB4lu3rX8iaSy5fALp8ZOWa8wVxALhbYsWiye6rf0DrdeYK0vd5r5N8PWWQLHQbfC7uMV7AgVau4zbHGlXAMtGOc39ohXWWGXsfotOvApYBaYh0TzyvaB3wVDTMCRETL2CFLhH2v4SfRXxIgjoGO4LZxraY/knGp0n2pZufTrptrs19ngnVZBdtrd0XxBQ82giF7i/hNWKbu4JTLKUcBPr6xCZM+mPF1LdhrclJM4kfkfoRArMUGq9Lbn9+e86aeWkwjWIr7Dkt1zdNw0OvZKCbO9DabCdPaYYnXPLQDKl0g3vc4YF9A1DM5A8oiuRsHx6csW9GHuNVr9FUPsOv0vZJYcUKHJdFr+P/+otduVIRdHojBPQGIY3LZ3BrMqW27S7xzo4DOYPn7B+WT8IX3lhosvYP3K/nfTzNdRU3KEElwp2rgF29T5SxV1teDcJh2ldYEALROzDD7ZFNdBUHCKiRN0NKCpNYLaGauhDqD1YKpjApNOIH3gLjQzFvZ5IAY0C+hRjBbKvaEBxcylQolSF4soPnESpEnpfsRvGNFTczQp+r+4dla/QbVs/0PuKgdxqokqUDiNkc+gcFEiitAn6iuQY5rxCyYi+x/swDF9h0Se/MIlSNS4pUOS6besHxX2JwLXrZRTbe6gaGho66l7pNq0fPHoIBxKyKapMhxF1K1LBkSRDgzgz6XYZycM53gkvqaF41aJ+XhEuxB5OouLYsLavCIu4emQFAL29r6WCg9+MHJhEj945Laq9zfdiFazcjaL1t6KjVJWaIUVzinLE14/RL5f8RUASC1LhtLzQ2Mmr28SRqLhM/yVkM6tnUzAvP5nyctMLJ/W/obwytyEjmllpkOay5Qgk0UgvNMfnNJTH5jOI71TRG/Ro5BrJO8BAnGJC+/v7QkP0YKwwBBrjjBJ4n2VEi9snSuRGV3bfemSIz/gIUhWtenjleP0HR143fUQZQcOl/f31wMIlSvtxshsOGXZfnwXayn9DKlWky4IjkzzOmOq2+h3oYrZD6BElNlANz4rq/J1VSP+eYR23eXREs5T/tNtglYBZdGm3nMMWLF1KdcuSAwsSqT1RHBxKBeJVLbR44rjED0qkVqLFE8dPjjiBzJPG5wBL5JCVwi0ecb4RLzBrMNtf/jEHvdqztUKIjIyMlgojmN1gjXYK93hu8IGiQP9BXAALpHa4ZbIFpJe40/yKc3TC9BJ3LPrQ4vaJOmB7iSqKsssrYFdDyVDconB5W1W3eT0wUtRDrWHd/AsuFZquUmQnUUae1I4WcLt5Gr8e2BzSBDTQluNV5uJu7g9K3wXv45Fzm+U+uAekGDmbND0BJmIYhmEYhmEYhmEYhmEYhmEY5j/FPx3KSD8WrnwpAAAAAElFTkSuQmCC"
      />
        
      <h5
        className="About__title__2"
      >
        About
      </h5>
    </div>
    <div
      className="About__content"
    >
      <p
        className="About__content__header"
      >
        Name
      </p>
      <p
        className="About__content__body"
      >
        anonymous
      </p>
    </div>
    <div
      className="About__content"
    >
      <p
        className="About__content__header"
      >
        Email
      </p>
      <p
        className="About__content__body"
      >
        no email
      </p>
    </div>
    <div
      className="About__content"
    >
      <p
        className="About__content__header"
      >
        Mobile Phone
      </p>
      <p
        className="About__content__body"
      >
        no mobile number
      </p>
    </div>
  </div>
</div>
`;
